setuptools/config/expand.py

```diff
@@ -359,7 +361,8 @@ def entry_points(text: str, text_source="entry-points") -> dict[str, dict]:
     entry-point names, and the second level values are references to objects
     (that correspond to the entry-point value).
     """
-    parser = ConfigParser(default_section=None, delimiters=("=",))  # type: ignore
+    # TODO: Explain why passing None is fine here when ConfigParser.default_section expects to be str
+    parser = ConfigParser(default_section=None, delimiters=("=",))  # type: ignore[call-overload]
```

Comment:
Could this be ?
```suggestion
    parser = ConfigParser(default_section="", delimiters=("=",))
```

Reply:
The meaning of `default_section` is [documented here](https://docs.python.org/3/library/configparser.html#customizing-parser-behaviour). It suggests that the default section could be overridden from the default of `"DEFAULT"`.

I don't know why the original author passed `default_section=None`. Reading briefly through the [configparser code](https://github.com/python/cpython/blob/main/Lib/configparser.py), it looks like the `default_section` is often compared for equality, so `None` is effectively treated as "no default". Presumably that was the intention of this usage.

Probably what should happen here is configparser should be explicit about what types are allowed here. I scanned through the configparser tests, but didn't see anything that explicitly allowed or excluded `None` as a valid type, and since `None` appears to be valid per this usage, it's probably more accurately an `Any` or `Hashable` type. 

My recommendation - file a ticket with python/cpython (or python/typeshed) indicating that the type spec for this parameter is ambiguous and should be refined, and then link that issue here instead of a TODO (since there's nothing "to do" in setuptools until such a refinement is present).

Reply:
The author being @abravalheri in https://github.com/pypa/setuptools/pull/3067/files#diff-2e8e2332e00f145022b07f1a5de895e643f38570a1765bd2da26e27fe29b64cdR305 I'll ask even if that was a couple years ago :)

Do you remember if there was any reason to set `ConfigParser`'s `default_section` to `None` instead of using the default of `"DEFAULT"` ? (re-exported as `configparser.DEFAULTSECT`)

Reply:
In the mean time, I marked setuptool's usage as undocumented behaviour, linking to his issue/question I created: https://github.com/python/typeshed/issues/12700

Further actions can come in a follow-up PR once the question is resolved and an action has been taken.

Reply:
> Do you remember if there was any reason to set `ConfigParser`'s `default_section` to `None` instead of using the default of `"DEFAULT"` ? (re-exported as `configparser.DEFAULTSECT`)

I believe I was trying to achieve the behaviour described by Jason, i.e. "no default".

Reply:
@abravalheri Thanks. This confirms what we thought. I think the new comment of calling this out as undocumented behaviour but not changing it is probably fine.